lib/src/rules/prefer_final_fields.dart

```diff
@@ -64,6 +64,27 @@ class GoodMutable {
 ```
 ''';
 
+bool _isMutated(
+        VariableDeclaration variable, CompilationUnit compilationUnit) =>
+    DartTypeUtilities
+        .traverseNodesInDFS(compilationUnit)
+        .where((n) =>
+            n is SimpleIdentifier && n.bestElement is PropertyAccessorElement)
+        .any((n) {
+      SimpleIdentifier identifier = n as SimpleIdentifier;
+      PropertyAccessorElement bestElement =
+          identifier.bestElement as PropertyAccessorElement;
+      if (bestElement.variable != variable.element) {
+        return false;
+      }
+
+      return identifier.getAncestor((a) =>
+              a is AssignmentExpression ||
```

Comment:
Kind of seems like we want to also ensure that the identifier is on the left-hand side of the assignment. For example, `var local; local = _field;` Doesn't mean that `_field` is modified. Similarly `array[_i] = 0;` doesn't mean that `_i` is modified.